dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaManagerHandle.java

```diff
@@ -41,10 +42,28 @@ public ClouderaManagerHandle(URI apiURI, CloseableHttpClient httpClient) {
     Preconditions.checkNotNull(apiURI, "Cloudera's apiURI can't null.");
     Preconditions.checkNotNull(httpClient, "httpClient can't be null.");
 
-    this.apiURI = apiURI;
+    // Always add trailing slash for safety
+    this.apiURI = unify(apiURI);
     this.httpClient = httpClient;
   }
 
+  /** 1. Remove query params and url fragments 2. Add trailing slash for safety */
+  private static URI unify(URI uri) {
```

Comment:
Should not this method be moved outside of Cloudera Handler. For example to `utils` package.

Reply:
No, removing params and fragments in unique for our use case with Cloudera. If we will find more usage in the future we will extract it. For now it's Cloudera internal invariant that we can rely on in Cloudera tasks.